wordpress-mobile/WordPress-iOS
# Blogging Prompts: Renew prompt reminders for current blog

Refs #18761
Depends on #18774

This adds logic that fetches prompts and renews the prompt notifications on app launch for the current blog. There's a check in place to ensure that the fetch only happens if the user has enabled prompt reminders for the current blog.

Note: In theory, it's possible for the user to have multiple sites with prompt reminders enabled, but I'm keeping the scope small for now; limited to the currently selected blog. This is so we don't have to make multiple network requests per Site since the endpoint doesn't support fetching prompts for multiple blogs at once.

Bonus: I've addressed some of the Xcode warnings highlighted by @leandroalonso here: https://github.com/wordpress-mobile/WordPress-iOS/pull/18773#issuecomment-1141412736

## To Test

Ensure that Blogging Prompt feature flag is enabled.

- Start with a site that has no prompt reminders scheduled.
- Verify that the schedule renewal is not called.
- Go to Site settings > Blogging reminders, and set a reminder schedule for Blogging Reminders.
- Kill the app, and open it up again.
- Verify that the schedule renew is not called.
- Go to Site Settings again, this time turning ON the "Include Prompts" switch.
- Kill the app, and open it up again.
- Verify that the schedule renewal is called.

## Regression Notes
1. Potential unintended areas of impact
Should be none.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Provided manual testing steps.

3. What automated tests I added (or what prevented me from doing so)
N/A.

PR submission checklist:
